packages/system/src/Fiber/context.ts

```diff
@@ -776,6 +777,20 @@ export class FiberContext<E, A> implements Fiber.Runtime<E, A> {
                   break
                 }
 
+                case "XPure": {
+                  const res: E.Either<any, any> = X.runEither(
+                    X.provideAll(this.environments?.value || {})(current)
+                  )
+
+                  if (res._tag === "Left") {
+                    current = fail.fail(res)[_I]
```

Comment:
should this be `res.left` passed to `fail.fail`?

Reply:
yes